File: backend/app/models/player.py
from typing import Annotated

from pydantic import BaseModel, Field

from app.models.position import Position

Direction = Annotated[float, Field(ge=0, lt=360)]


class Player(BaseModel):
    id: str = Field(min_length=1)
    name: str = Field(min_length=1)
    number: int = Field(gt=0, le=99)
    team_id: str = Field(min_length=1)
    position: Position
    orientation: Direction

File: frontend/src/models/player.ts
import { Position } from "./position";

export type PlayerSpeedCategory = "BASELINE" | "FAST" | "SUPER_FAST";

export type Player = {
  id: string;
  name: string;
  number: number;
  teamId: string;
  position: Position;
  orientation: number;
  speedCategory: PlayerSpeedCategory;
  // Velocity in centimeters per second while animation playback is active.
  velocity?: Position;
};
